tests/common/startup/jormungandr_starter.rs

```
@@ -0,0 +1,102 @@
+#![cfg(feature = "integration-test")]
+
+extern crate custom_error;
+
+use self::custom_error::custom_error;
+
+use common::configuration::jormungandr_config::JormungandrConfig;
+use common::jcli_wrapper;
+use common::jormungandr_wrapper;
+use common::process_utils;
+use common::process_utils::{
+    output_extensions::ProcessOutput, process_guard::ProcessKillGuard, ProcessError,
+};
+
+use std::process::{Command, Output};
+
+custom_error! {pub StartupError
+    JormungandrNotLaunched{ source: ProcessError } = "could not start jormungandr",
+}
+
+fn try_to_start_jormungandr_node(
+    rest_address: &str,
+    command: &mut Command,
+) -> Result<ProcessKillGuard, StartupError> {
+    println!("Starting jormungandr node...");
+    let process = command
+        .spawn()
+        .expect("failed to execute 'start jormungandr node'");
+
+    let guard = ProcessKillGuard::new(process, String::from("Jormungandr node"));
+
+    let proces_start_result = process_utils::run_process_until_response_matches(
+        jcli_wrapper::jcli_commands::get_rest_stats_command(&rest_address),
+        &is_node_up,
+        2,
+        5,
+        "get stats from jormungandr node",
+        "jormungandr node is not up",
+    );
+
+    match proces_start_result {
+        Ok(_) => return Ok(guard),
+        Err(e) => return Err(StartupError::JormungandrNotLaunched { source: e }),
+    }
+}
+
+fn start_jormungandr_node_sync_with_retry(
+    rest_address: &str,
+    command: &mut Command,
+    config: &mut JormungandrConfig,
+) -> ProcessKillGuard {
+    let first_attempt = try_to_start_jormungandr_node(rest_address, command);
+    match first_attempt {
+        Ok(guard) => return guard,
+        _ => (),
```

Might want to report the failure?